docs/overview/architecture.md

```diff
@@ -0,0 +1,237 @@
+    - [Vald Meta Gateway](#vald-meta-gateway)
+    - [Vald Meta](#vald-meta)
+  - [Vald Backup](#vald-backup)
+    - [Vald Compressor](#vald-compressor)
+    - [Vald Backup Manager](#vald-backup-manager)
+    - [Vald Backup Gateway](#vald-backup-gateway)
+  - [Vald Load Balancing](#vald-load-balancing)
+    - [Vald LB Gateway](#vald-lb-gateway)
+    - [Agent Discoverer](#agent-discoverer)
+  - [Vald Core Engine](#vald-core-engine)
+    - [Vald Agent](#vald-agent)
+    - [Vald Agent Scheduler](#vald-agent-scheduler)
+    - [Vald Index Manager](#vald-index-manager)
+  - [Vald Replication Manager](#vald-replication-manager)
+    - [Vald Replication Manager Agent](#vald-replication-manager-agent)
+    - [Vald Replication Manager Controller](#vald-replication-manager-controller)
+  - [Kubernetes Components](#kubernetes-components)
+    - [Kube-apiserver](#kube-apiserver)
+    - [Custom Resources](#custom-resources)
+
+## Overview
+
+
+
+Vald is based on [Kubernetes](https://kubernetes.io/) architecture.
+Before you read this document, you must understand the basic concept of Kubernetes.
+
+The below image is Vald's architecture.
+
+<img src="../../design/Vald Future Architecture Overview.svg" />
+
+We will explain this image in the following section.
+
+## Data Flow
+
+In this section, we will explain the data flow in Vald which is the most important for users.
+
+### Insert
+
+<img src="../../assets/docs/insert_flow.svg" />
+
+When the user inserts data into Vald:
+
+1. Vald Ingress receives the request from the user. The request includes the vector and the vector ID which is set by user.
+2. Vald Ingress will forward the request to Vald Meta Gateway. After the Vald Meta Gateway receive the request, the UUID will be generated for internal use.
+3. Vald Meta Gateway will forward the request with UUID to Vald Backup Gateway.
+4. Vald Backup Gateway will forward the request to Vald LB Gateway.
+5. Vald LB Gateway will decide which Vald Agent instance to process the request based on the node resource usage and forward the request to the decided Vald Agent.
+6. Vald Agent returns success and the corresponding Vald Agent IP address to Vald LB Gateway.
+7. Vald LB Gateway returns success and the IP address to Vald Backup Gateway.
+8. Vald Backup Gateway returns success to Vald Meta Gateway.
+9. Vald Meta Gateway will send the metadata including the UUID and the vector ID to the Vald Meta.
+10. Vald Meta will store the metadata to the persistent layer.
+11. Vald Backup Gateway will send all the data (vector, vector ID and UUID) the Vald Compressor.
+12. Vald Compressor will compress the data received and send to the Vald Backup Manager.
+13. Vald Backup Manager will store the compressed data to the persistent layer.
+
+### Search
+
+<img src="../../assets/docs/search_flow.svg" />
+
+When the user searches a vector from Vald:
+
+1. Vald Ingress receives the search request from the user. The request includes the vector or the vector ID.
```

Comment:
```suggestion
1. Vald Ingress receives a search request from the user, Vald has a Search interface to search by vector and a SearchByID interface to search by stored vector index IDs, so that the request can be searched using a vector or vector ID.
```